src/utils/moreurls/params.go

```diff
@@ -0,0 +1,104 @@
+package moreurls
+
+import (
+	"github.com/rs/zerolog/log"
+
+	"github.com/hearchco/agent/src/utils/morestrings"
+)
+
+// Params struct, slice of Param struct.
+type Params struct {
+	params []Param
+}
+
+// Constructs a new slice of params with provided keys and values.
+// Input should be in pairs: "key1, value1, key2, value2, key3, value3, ..."
+// Number of elements must be even, otherwise this function panics.
+func NewParams(elem ...string) Params {
+	if len(elem)%2 != 0 {
+		log.Panic().
+			Strs("elem", elem).
+			Msg("Odd number of elements for KV pairs")
+		// ^PANIC - Assert even number of elements.
+	}
+
+	// Extract keys and values from elements.
+	length := len(elem) / 2
+	keys := make([]string, 0, length)
+	values := make([]string, 0, length)
+	for i, e := range elem {
+		if i%2 == 0 {
+			keys = append(keys, e)
+		} else {
+			values = append(values, e)
+		}
+	}
+
+	// Create parameters slice.
+	p := make([]Param, 0, length)
+	for i := range length {
+		p = append(p, Param{keys[i], values[i]})
+	}
+
+	return Params{p}
+}
+
+// Returns the value of the first occurence with the provided key.
+// If not found, returns empty string and false.
+func (p Params) Get(k string) (string, bool) {
+	for _, param := range p.params {
+		if param.Key() != k {
+			continue
+		}
+
+		return param.Value(), true
+	}
+
+	return "", false
+}
+
+// Sets the value to the first occurence of the provided key.
+// If not found, appends new Param KV pair and returns false.
+func (p *Params) Set(k, v string) bool {
+	for i, param := range p.params {
+		if param.key != k {
```

Comment:
param.Key* Probably move these into separate packages to have proper private fields working